mono/mini/interp-stubs.c

```diff
@@ -149,32 +149,18 @@ stub_delegate_ctor (MonoObjectHandle this_obj, MonoObjectHandle target, gpointer
 void
 mono_interp_stub_init (void)
 {
-	if (mini_get_interp_callbacks ()->create_method_pointer)
+	if (mini_get_interp_callbacks ())
 		/* already initialized */
 		return;
 
-	MonoEECallbacks c;
-	c.create_method_pointer = stub_create_method_pointer;
-	c.create_method_pointer_llvmonly = stub_create_method_pointer_llvmonly;
-	c.runtime_invoke = stub_runtime_invoke;
-	c.init_delegate = stub_init_delegate;
-	c.get_remoting_invoke = stub_get_remoting_invoke;
-	c.set_resume_state = stub_set_resume_state;
-	c.run_finally = stub_run_finally;
-	c.run_filter = stub_run_filter;
-	c.frame_iter_init = stub_frame_iter_init;
-	c.frame_iter_next = stub_frame_iter_next;
-	c.find_jit_info = stub_find_jit_info;
-	c.set_breakpoint = stub_set_breakpoint;
-	c.clear_breakpoint = stub_clear_breakpoint;
-	c.frame_get_jit_info = stub_frame_get_jit_info;
-	c.frame_get_ip = stub_frame_get_ip;
-	c.frame_get_arg = stub_frame_get_arg;
-	c.frame_get_local = stub_frame_get_local;
-	c.frame_get_this = stub_frame_get_this;
-	c.frame_get_parent = stub_frame_get_parent;
-	c.start_single_stepping = stub_start_single_stepping;
-	c.stop_single_stepping = stub_stop_single_stepping;
-	c.delegate_ctor = stub_delegate_ctor;
+#define stub_entry_from_trampoline NULL
```

Comment:
This isn't ideal, in that, it "works" even if there is a function. It'll just replace it.
Maybe better to provide a function with unreachable?

Reply:
`#define stub_entry_from_trampoline g_assert_not_reached`?

Reply:
`g_assert_not_reached` is a macro, can't take its address, has to be wrapped in a function.
That is larger than just `NULL` but I think preferable.
Because if you actually write it twice, it will fail to compile.
Whereas the define will silently replace a function if it occurs after it (or error if it occurs before it).

It'd be nice to have a typed null constant variable like:

```
void * const a = 0; // but actual function pointer type
struct 
{
 void * b; // but actual type
} c = { a };
```

but that isn't legal.

So  I think provide the little extra stub functions, that `g_assert_not_reached`.
Or `g_assertf (0, "");` which includes the function name, but `g_assert_not_reached` does not.
I'll amend the PR as such shortly.